truth_set/java/structure/organisation/statement-issues/OrderProcessor.java

```diff
@@ -0,0 +1,79 @@
+import java.util.List;
+import java.util.Arrays;
+
+class Item {
+    private String name;
+    private double price;
+    private int quantity;
+
+    public Item(String name, double price, int quantity) {
+        this.name = name;
+        this.price = price;
+        this.quantity = quantity;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+}
+
+class Order {
+    private String id;
+    private List<Item> items;
+    private String discountCode;
+
+    public Order(String id, List<Item> items, String discountCode) {
+        this.id = id;
+        this.items = items;
+        this.discountCode = discountCode;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public List<Item> getItems() {
+        return items;
+    }
+
+    public String getDiscountCode() {
+        return discountCode;
+    }
+}
+
+public class OrderProcessor {
+    public static double processOrder(Order order, double taxRate) {
+        double total = (calculateItemsTotal(order.getItems()) - 
+                      (order.getDiscountCode() != null ? 50 : 0)) * 
+                      (1 + taxRate);
+        
+        System.out.printf("Order %s processed. Total after tax: $%.2f%n", 
+                         order.getId(), total);
+        return total;
+    }
+
+    private static double calculateItemsTotal(List<Item> items) {
+        double sum = 0;
+        for (Item item : items) {
+            sum += item.getPrice() * item.getQuantity();
+        }
+        return sum;
+    }
+
+    public static void main(String[] args) {
+        List<Item> items = Arrays.asList(
+            new Item("Laptop", 999.99, 1),
+            new Item("Mouse", 49.99, 2),
+            new Item("Keyboard", 79.99, 1)
+        );
+        
+        Order order = new Order("ORD123", items, "SUMMER21");
+        double taxRate = 0.08;
+        double total = processOrder(order, taxRate);
+        System.out.println(total);
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider moving test code to unit tests</b></div>
<div id="fix">

The `main` method contains test-like code that should be moved to proper unit tests
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2530735434>#cc25d3</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged